File: content/DP/lab2/src/main/java/ru/prokdo/model/file/FileWriter.java
package ru.prokdo.model.file;

import java.nio.file.Paths;
import java.io.BufferedWriter;
import java.nio.file.Files;


public final class FileWriter {
    public final String filePath;

    public FileWriter(String filePath) {
        if (filePath == null) throw new IllegalArgumentException("File path cannot be null");
        
        this.filePath = filePath;
    }

    public boolean write(String str) {
        if (str == null || str.isBlank() || str.isEmpty()) return false;

        try { 
            BufferedWriter writer = Files.newBufferedWriter(Paths.get(filePath)); 

            writer.write(str);

            writer.close();

            return true;
        }
        catch (Exception exception) { return false; }
    }
}

File: content/DP/lab3/src/main/kotlin/model/file/FileWriter.kt
package model.file

import java.io.File

object FileWriter {
    fun write(file: File, content: List<String>): Boolean {
        try { file.writeText("") }
        catch (ignored: Exception) { return false }

        try { content.forEach { file.appendText(it.toString() + "\n") } }
        catch (ignored: Exception) { return false }

        return true
    }
}